src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs

```diff
@@ -160,8 +168,31 @@ private async Task GetSuggestionsTaskAsync(bool isAutomaticOnInitialization, Can
         {
             return;
         }
-        _ = await _smartRenameSession.GetSuggestionsAsync(cancellationToken).ConfigureAwait(true);
-        return;
+
+        if (IsUsingContext)
+        {
+            var document = this.BaseViewModel.Session.TriggerDocument;
+            var smartRenameContext = ImmutableDictionary<string, string[]>.Empty;
+            var editorRenameService = document.GetRequiredLanguageService<IEditorInlineRenameService>();
+            if (editorRenameService.IsRenameContextSupported)
+            {
+                var renameLocations = await this.BaseViewModel.Session.AllRenameLocationsTask.JoinAsync(cancellationToken)
+                    .ConfigureAwait(true);
+                var context = await editorRenameService.GetRenameContextAsync(this.BaseViewModel.Session.RenameInfo, renameLocations, cancellationToken)
+                    .ConfigureAwait(true);
+                smartRenameContext = ImmutableDictionary.CreateRange<string, string[]>(
+                    context
+                    .Select(n => new KeyValuePair<string, string[]>(n.Key, n.Value.ToArray())));
+            }
+
+            _ = await _smartRenameSession.GetSuggestionsAsync(smartRenameContext, cancellationToken)
+                .ConfigureAwait(true);
```

Comment:
i'm a bit confused by this.  we get the suggestions... but do nothing with them?

Reply:
`_smartRenameSession.GetSuggestionsAsync` would updated the binded property in this view model, and cause the UI to show the name suggestions.